flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/source/reader/IcebergSourceSplitReader.java

```diff
@@ -80,6 +87,25 @@ public RecordsWithSplitIds<RecordAndPosition<T>> fetch() throws IOException {
     }
 
     if (currentReader.hasNext()) {
+      // Wait until the reader is blocked. Wake every second in case this missed a signal
+      boolean first = true;
+      while (pausedSplits.contains(currentSplitId)) {
+        if (first) {
+          LOG.info("Paused reading {}", currentSplitId);
```

Comment:
all logs should include subtask index as important context

Reply:
The subtask index is usually included in the thread name, but added it anyway